packages/app-cli/tests/command-mkbook.ts

```diff
@@ -0,0 +1,43 @@
+import { execCommand2 } from '@joplin/tools/tool-utils';
+import { homedir } from 'os';
+import * as fsE from 'fs-extra';
+
+const profileDir = `${homedir()}/.config/joplindev-populate/joplindev-testing-command-mkbook`;
+fsE.removeSync(profileDir);
+
+// eC2W is an wrapper for execCommand2
+const eC2W = async (command: string) => {
+	await execCommand2(`yarn start-no-build --profile ${profileDir} ${command}`, { quiet: true });
+};
+
+describe('CreateSubFolder', function() {
+
+	test('Create folders', (async () => {
+		await eC2W('mkbook test1');
+		await eC2W('mkbook test2');
+	}));
+
+	test('Create sub-folder', (async () => {
+		await eC2W('use test1');
+		await eC2W('mkbook -s test1.1');
+	}));
+
+	test('Create sub-folder in target folder', (async () => {
+		await eC2W('mkbook -s test2.1 test2');
+	}));
```

Comment:
Those aren't tests since you're not checking the output. Also I'm not keen on running the CLI on tests as it's very slow. I think the test framework should be setup so that we can call a command directly by code, for example by creating a `new Command()` instance and running `command.action(args)` directly. There might a bit of work to setup the environment but that would be overall much better than running a slow CLI on each test.

Reply:
@laurent22 I have improved the tests independently of your statement. 

I agree with you that a slow CI is a pain.
Unfortunately I have limited knowledge of js/ts ecosystem and how it can be solved the way you describe. 
But I can offer some solutions how I could try to solve it.

Solution 1: I let the tests out, the make command is small and if someone will review it and accept the change so we can move on. Additionally to it i let the test code in the repo with documentation about your test framework point or reverent it in the merge commit! (I am honestly, i am not really happy about this solution, because this could possibly lead to problems in the future.)

Solution 2: Getting the "cli-integration.sh" working again. If someone makes changes, they had to execute the script which triggers the tests cli commands.

Reply:
I've now setup the app-cli package to allow testing commands. Have a look at https://github.com/laurent22/joplin/blob/dev/packages/app-cli/app/command-done.test.ts for example to see how it works. You'll just need to copy the setup and write your test from this. Please let me know if something's not clear or not working.

Reply:
Thanks, i have implement tests with the new framework, should be done now. - One small note to the [commit](https://github.com/laurent22/joplin/commit/7992fe5b634a36c38dfa61a4a0d11150ff87ecaa) it leads to errors in many functions for example "exit", "mv" and so one, presumably because of the changed line `const BaseCommand = require('./base-command').default;`